### DISCUSSION TITLE
Remove or add all Rush Plus services?

### DISCUSSION THREAD

**Discussion by rhwood** [emma-k-alexandra/WMATA.swift#29] [Ideas]
So I found a 2012 Metro Pocket Guide [on Ebay](https://www.ebay.com/itm/123718657520) that depicts the Orange Line Rush Plus service to Largo Town Center:

![s-l1600](https://user-images.githubusercontent.com/297232/147789697-99ee0cb3-dd4c-44a8-b1b5-d018eeacefe3.jpg)

Since the WMATA API does not return either Rush Plus service as a separate line (and when implemented, the trains merely had "Yellow" or "Orange" on the front and the alternate destination on the side), should Yellow Line Rush Plus simply be removed from this library as a thing?

**Comment by emma-k-alexandra** [comment]
> Since the WMATA API does not return either Rush Plus service as a separate line

Oh, I wish this was true. While working v11 I contacted WMATA asking if Yellow Line Rush Plus was still returned in the API. The Manager of Application Systems responded & told me YLRP isn't returned in the API anymore. Unfortunately, during my testing I discovered that the [Standard Routes](https://developer.wmata.com/docs/services/5763fa6ff91823096cac1057/operations/57641afc031f59363c586dca) endpoint does, indeed, return `YLRP`. Here's a response:

```json
{
    "StandardRoutes": [
        ...
        {
            "LineCode": "YLRP",
            "TrackNum": 1,
            "TrackCircuits": [...]
        }
        ...
    ]
}
```

I responded to WMATA's API support informing them that this endpoint still returns `YLRP` and they indicated this part of the API is in need of an update.

Since the API is still returning this value, I felt the need to support it. There's definitely a lot of `Line` grossness that happens because of `YLRP` though. I honestly doubt `YLRP` routing information is very useful to most devs. I can think of two ways to remove it

1. Map `YLRP` to `nil` in the Standard Routes response. This would result in `Rail.StandardRoutes.Response.StandardRoute.line` being optional.
2. Attempt to remove the `YLRP` element from the Standard Routes response during decoding

Solution 1 is pretty easy, since we have `@MapToNil`. I'm not sure how solution 2 would work, it could get a bit hairy but would provide a better user experience which is nice.

Either way, I think removing YLRP would probably be beneficial. I was hoping I could convince WMATA themselves to do it, but if we need to do it within this package I'd be okay with that too. I think either solution is better than adding all of the RP lines 😁

**Reply by rhwood** [reply]
Well drat. I didn't notice that. BTW, I think that the Orange Line Rush Plus service got replaced by the Silver Line before the StandardRoutes method appeared in the API, which is why it wasn't present.

**Comment by emma-k-alexandra** [comment]
@rhwood Good news! WMATA has updated the Standard Routes endpoint and removed `YLRP`. It can be removed!

**Reply by rhwood** [reply]
> @rhwood Good news! WMATA has updated the Standard Routes endpoint and removed `YLRP`. It can be removed!

🎉 (I was just writing the same. I asked WMATA last week why it was included...and just got an email from them confirming its removal.)

**Comment by emma-k-alexandra** [comment]
Removed YLRP in v13, available now
